pkg/template/app/app.go

```diff
@@ -41,25 +46,38 @@ type AppCROutput struct {
 
 func NewAppCR(config Config) ([]byte, error) {
 	userConfig := applicationv1alpha1.AppSpecUserConfig{}
-
-	var namespace string
+	appLabels := map[string]string{}
+
+	// Accomodating all the label cases here:
+	// 1. In-cluster Apps get unique label
+	// 2. Org-namespaced Apps get cluster label
+	// 3. Cluster-namespaced Apps with defaulting disabled gets default version label
+	// 4. Cluster-namespaced Apps with defaulting enabled gets nothing
+	var crNamespace string
 	if config.InCluster {
-		namespace = config.Namespace
+		crNamespace = config.Namespace
+		appLabels[label.AppOperatorVersion] = "0.0.0"
+	} else if config.Organization != "" {
+		crNamespace = fmt.Sprintf("org-%s", config.Organization)
+		appLabels[label.Cluster] = config.Cluster
+	} else if !config.DefaultingEnabled {
+		appLabels[label.AppOperatorVersion] = "1.0.0"
```

Comment:
There is some technical debt here :( `1.0.0` was used for app CRs using helm 2 and is now no longer in use.

If you apply this does app-admission-controller update the value to the correct version for the cluster?

Reply:
Let me check that, I haven't tried to apply it live yet

Reply:
@rossf7 update from my side, when I submitted app with `1.0.0` label it was updated to the correct value. If that's the case I guess we can get rid of this part of labeling logic, WDYT?

Reply:
@ljakimczuk Thanks for checking and yes let's remove the version label here so its defaulted.

We don't want customers adding it to their gitops repo as its an implementation detail. There is an `appversionupdate` resource in cluster-operator that ensures the label value is set correctly.

Reply:
Done, we no longer set the `1.0.0` version